pkg/cmd/grafana-cli/commands/secretsmigrations/reencrypt_secrets.go

```diff
@@ -0,0 +1,207 @@
+package secretsmigrations
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/runner"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/utils"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
+	"github.com/grafana/grafana/pkg/services/secrets"
+	"github.com/grafana/grafana/pkg/services/secrets/manager"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"xorm.io/xorm"
+)
+
+type simpleSecret struct {
+	tableName       string
+	columnName      string
+	isBase64Encoded bool
+}
+
+func (s simpleSecret) reencrypt(secretsSrv *manager.SecretsService, sess *xorm.Session) error {
+	var rows []struct {
+		Id     int
+		Secret string
+	}
+
+	selectSQL := fmt.Sprintf("SELECT id, %s as secret FROM %s", s.columnName, s.tableName)
```

Comment:
My biggest question with putting code that interacts directly with the database into commands is whether it would be better placed in the service itself so that it can be re-used rather than being locked up in the cli.  I'm thinking for example of the plugin management code that had to be moved out to support the catalog.

Reply:
The other thing I notice here is that we're building SQL with Sprintf, which is generally an anti-pattern due to the potential for SQL injection.  In this case we're not taking input from the user so it's not a potential vector, but it's a pattern we should avoid if we can.  I'm not sure off-hand whether xorm supports parameterization of identifiers like column and table names, but it might be worth checking into that as an alternative approach.

Reply:
> My biggest question with putting code that interacts directly with the database into commands is whether it would be better placed in the service itself so that it can be re-used rather than being locked up in the cli. I'm thinking for example of the plugin management code that had to be moved out to support the catalog.

I generally would agree, but I think it does not exactly apply for this specific case because (especially in large setups) the execution of this command would case database locks and impact the users performance, so I think it could make sense to keep it as a CLI command explicitly and only recommend it for very specific situations and probably with a derived downtime. 

Additionally, refactoring this command as-is into the `secrets.Service`, for instance, would be easy even if that happens later, so considering the time constraints I'll keep it here, while taking a final decision, and move it later, if so. Consider that we'll need to work on some other secrets-related commands, so would be pretty easy to do it later if actually required.

> The other thing I notice here is that we're building SQL with Sprintf, which is generally an anti-pattern due to the potential for SQL injection. In this case we're not taking input from the user so it's not a potential vector, but it's a pattern we should avoid if we can. I'm not sure off-hand whether xorm supports parameterization of identifiers like column and table names, but it might be worth checking into that as an alternative approach.

Sure! As you mentioned, it's not a potential vector. But yeah, let me find a quick workaround.

Reply:
Hey @DanCech! I've spent some time on exploring XORM docs but I haven't been able to replace all the `fmt.Sprintf` usages. Anyway, I replaced some of those usages by function calls like `Table`, `Where`, `Cols` but I'm not 100% sure these are safe neither. It uses args (`?`) as most ORMs to avoid SQLi but these are not supported to specify table & column names. I suspect because these are not common vectors of SQLi. In fact, I already had in mind this common pattern when using ORMs so that's why I was already using these args when possible.

Any thoughts? Would it be acceptable to merge it as-is for you? Otherwise, what do you suggest as next step? Thanks!

Reply:
I think that's a reasonable compromise, some systems support alternative placeholder syntax (eg `??`) for column/table names to get around this problem but it doesn't seem like xorm does.